testing/test_assertion.py

```diff
@@ -813,15 +813,14 @@ def test_recursive_dataclasses_verbose(self, testdir):
                 "*Matching attributes:*",
                 "*['field_a']*",
                 "*Differing attributes:*",
-                "*field_b: ComplexDataObject2(*SimpleDataObject(field_a=2, field_b='c')) != ComplexDataObject2(*SimpleDataObject(field_a=3, field_b='c'))*",  # noqa
-                "*Matching attributes:*",
-                "*['field_a']*",
-                "*Differing attributes:*",
-                "*field_b: SimpleDataObject(field_a=2, field_b='c') != SimpleDataObject(field_a=3, field_b='c')*",  # noqa
+                "*field_b=SimpleDataObject(field_a=2, field_b='c')) != ComplexDataObject2(field_a=ComplexDataObject*",
```

Comment:
Any reason why you truncated the line here? The test should preferably include the entire line.

Reply:
The reason was, that when I did a multiline string (to not keep it in one line to satisfy requirements of the formatter), it included the whitespace characters from the beginning of the line. E.g.:
When I did from:
`"*field_b: SimpleDataObject(field_a=2, field_b='c') != SimpleDataObject(field_a=3, field_b='c')*"`
the following:
`"*field_b: SimpleDataObject(field_a=2, field_b='c') != \
SimpleDataObject(field_a=3, field_b='c')*"`
The output at the ends was something like:
`"*field_b: SimpleDataObject(field_a=2, field_b='c') !=                                                    SimpleDataObject(field_a=3, field_b='c')*"`,
which made the mapping not possible.

Reply:
Hmmm we should probably avoid splitting the discussion over two PRs (one closed even at that).

Instead of continuing to reply here, how about copying the original message as a "quote" and link to the comment as a new post in the new PR?

Reply:
> to not keep it in one line to satisfy requirements of the formatter

So the preferable way to handle it in this case is to just tell the linter/formatter "I know what I'm doing".

For flake8, you do this with `# noqa` comments.

For black (the formatter), you do this by delineating the code black as not auto-formatted:

```py
# fmt: off
... Your code here -- will not be auto formatted ...
# fmt: on
```

Reply:
> Instead of continuing to reply here, how about copying the original message as a "quote" and link to the comment as a new post in the new PR?

TBH I am too lazy to do that :sleeping:

Reply:
> to not keep it in one line to satisfy requirements of the formatter

So the preferable way to handle it in this case is to just tell the linter/formatter "I know what I'm doing".

For flake8, you do this with `# noqa` comments.

For black (the formatter), you do this by delineating the code black as not auto-formatted:

```py
# fmt: off
... Your code here -- will not be auto formatted ...
# fmt: on
```